### DISCUSSION TITLE
How to add Azure AD group or M365 group to SharePoint group?

### DISCUSSION THREAD

**Discussion by sachinsatav** [pnp/pnpcore#783] [Q&A] [answered]
Hi,
I am trying to add an existing Azure AD group & M365 group to SharePoint group but I am struggling to find anything that explains how I can do that. 

`var siteGroup = await newSiteContext.Web.SiteGroups.FirstOrDefaultAsync(g => g.LoginName == "some group");`
`await siteGroup.AddUserAsync("someuser@domain.com");`

Using above I can add users to the group however not sure how can I add AD group or M365 group. 

I have tried - 

`await siteGroup.AddUserAsync("GROUP_NAME@domain.com"); //m365 group`
`await siteGroup.AddUserAsync("GROUP_NAME"); //Azure AD group`

but it did not work.

If anyone can help me with it, that would be appreciated. Thanks in advance.

**Comment by jansenbe** [comment, marked as answer]
@sachinsatav : adding Azure AD security groups or members of the Microsoft 365 groups is possible, currently you'll need to correctly format the loginName. Below samples should help, the GUIDs are the object ids of the respective Azure AD objects.

```csharp
// Add an Azure AD security group
await addedGroup.AddUserAsync("c:0t.c|tenant|c8e69a24-a625-48d7-b80b-c9656b9d4030");

// Add an Microsoft 365 group (members)
await addedGroup.AddUserAsync("c:0o.c|federateddirectoryclaimprovider|45dfc99d-a4a5-419d-ad79-013eed517fb2");
```

I'll add this information to our documentation, thanks for raising this.

**Reply by jansenbe** [reply]
Updated docs: https://pnp.github.io/pnpcore/using-the-sdk/security-users.html#sharepoint-users

![image](https://user-images.githubusercontent.com/7451219/157498445-e089647c-a671-45df-8652-5e2035ca1c59.png)

**Reply by sachinsatav** [reply]
Thanks @jansenbe. I have marked this as answer.
